Source/NSURL.m

```diff
@@ -2116,17 +2116,26 @@ + (instancetype)queryItemWithName:(NSString *)name
   return AUTORELEASE(newQueryItem);
 }
 
-- (instancetype)initWithName:(NSString *)name
-                       value:(NSString *)value
+- (instancetype) init
```

Comment:
I would prefer an "init" method that calls "initWithName: nil value: nil". The later setting name to @"" if it is nil. But your code is correct as well.

Reply:
Did the way you suggested.